libs/blocks/graybox-promote/components/graybox-promote-lit.js

```diff
@@ -114,7 +113,25 @@ const getSharepointData = async (url) => {
   };
 };
 
-const getProjectExcelPath = (url) => url.pathname.replace('.json', '.xlsx');
+const getFilePath = async ({ ref, repo, owner, excelRef } = {}) => {
+  try {
+    const status = await fetch(`${ADMIN}/status/${owner}/${repo}/${ref}?editUrl=${excelRef}`);
+    if (!status.ok) throw new Error('Failed to fetch file path');
+    return (new URL((await status.json())?.preview?.url)).pathname;
+  } catch (e) {
+    throw new Error(`Error getting file path: ${e.message}`);
+  }
+};
```

Comment:
I'm a huge fan of top level catching for async/await functions. The only advantage that you get by adding "manual" error handling on the function level here, is improved error messages - especially since you re-throw the error.

That IMO isn't worth obscuring the function. I'd go with a clean version of the function and handle the error a level above:

```suggestion
const getFilePath = async ({ ref, repo, owner, excelRef } = {}) => fetch(`${ADMIN}/status/${owner}/${repo}/${ref}?editUrl=${excelRef}`)
.then(res => res.json())
.then(json => new URL(json.preview.url).pathname)
```

Reply:
Got your point 👍 I simplified the function by removing the try catch [as the task will catch the errors](https://github.com/adobecom/milo/pull/2555/files#diff-7214629f44f7a1dbc0bdfc18fb6309965da48000fe961e4aaeed56046e9aa06fR290), but I would keep the `status.ok` check at least until this feature gets more stable. In this way we can debug easier in this incipient state